
For dummy connections, use the listener specified first in the
configuration - by default this will be port 80 rather than the SSL
port.

--- httpd-2.0.49/server/mpm_common.c.pod
+++ httpd-2.0.49/server/mpm_common.c
@@ -357,6 +357,7 @@
 AP_DECLARE(apr_status_t) ap_mpm_pod_open(apr_pool_t *p, ap_pod_t **pod)
 {
     apr_status_t rv;
+    ap_listen_rec *lr;
 
     *pod = apr_palloc(p, sizeof(**pod));
     rv = apr_file_pipe_create(&((*pod)->pod_in), &((*pod)->pod_out), p);
@@ -367,8 +368,11 @@
     apr_file_pipe_timeout_set((*pod)->pod_in, 0);
     (*pod)->p = p;
 
-    apr_sockaddr_info_get(&(*pod)->sa, ap_listeners->bind_addr->hostname,
-                          APR_UNSPEC, ap_listeners->bind_addr->port, 0, p);
+    /* Find the listener specified first in the configuration. */
+    for (lr = ap_listeners; lr->next != NULL; lr = lr->next)
+        /* noop */;
+
+    (*pod)->sa = lr->bind_addr;
 
     /* close these before exec. */
     apr_file_unset_inherit((*pod)->pod_in);
